openwisp_firmware_upgrader/upgraders/openwrt.py

```diff
@@ -408,24 +417,31 @@ def _refresh_addresses(self):
         handle cases in which the IP has changed
         """
         self.connection.device.refresh_from_db()
-        self.connection.refresh_from_db()
+        self.connection = self.connection.get_working_connection(self.connection.device)
         self.addresses = self.connection.get_addresses()
 
     def _write_checksum(self, checksum):
         for attempt in range(1, self.RECONNECT_MAX_RETRIES + 1):
-            self._refresh_addresses()
-            addresses = ', '.join(self.addresses)
-            self.log(
-                _(
-                    'Trying to reconnect to device at {addresses} (attempt n.{attempt})...'.format(
-                        addresses=addresses, attempt=attempt
-                    )
-                ),
-                save=False,
-            )
             try:
+                self._refresh_addresses()
+                addresses = ', '.join(self.addresses)
+                self.log(
+                    _(
+                        'Trying to reconnect to device at {addresses} (attempt n.{attempt})...'.format(
+                            addresses=addresses, attempt=attempt
+                        )
+                    ),
+                    save=False,
+                )
                 self.connect()
```

Comment:
shouldn't we update this too?

Reply:
I added the following wrapper methods so we don't need to reference the connection object again and again 

```
https://github.com/openwisp/openwisp-firmware-upgrader/blob/82ade68a1ba8c6cc988dc2a73e577e9abcf6d0e1/openwisp_firmware_upgrader/upgraders/openwrt.py#L123-L130
```